agent/consul/acl.go

```diff
@@ -227,22 +261,57 @@ func (c *aclCache) lookupACL(id, authDC string) (acl.ACL, error) {
 		reply.TTL = c.config.ACLTTL
 		reply.Parent = parent
 		reply.Policy = policy
-		return c.useACLPolicy(id, authDC, cached, &reply)
+		theACL, theError := c.useACLPolicy(id, authDC, cached, &reply)
+		if cached != nil && theACL != nil {
+			cached.ACL = theACL
+			cached.ETag = reply.ETag
+			cached.Expires = time.Now().Add(c.config.ACLTTL)
+		}
+		c.fireResult(id, theACL, theError)
+		return
 	}
 
 ACL_DOWN:
 	// Unable to refresh, apply the down policy.
 	switch c.config.ACLDownPolicy {
 	case "allow":
-		return acl.AllowAll(), nil
+		c.fireResult(id, acl.AllowAll(), nil)
+		return
+	case "async-cache":
```

Comment:
Shouldn't this be like this (or am I missing what you are trying to do here)

```
case "async-cache":
   fallthrough
case "extend-cache":
...
```

Reply:
Yes, you are right, wrong semantics coming from C... I did put case "async-cache", "extend-cache" everywhere in the next patch